chainer/functions/array/transpose_sequence.py

```diff
@@ -0,0 +1,63 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _transpose(xs, length):
+    xp = cuda.get_array_module(*xs)
+    lengths = numpy.zeros(length, dtype='i')
+    for i, x in enumerate(xs):
+        lengths[0:len(x)] = i + 1
+    dtype = xs[0].dtype
+    unit = xs[0].shape[1:]
+    outs = tuple(xp.empty((l,) + unit, dtype=dtype) for l in lengths)
```

Comment:
How about `outs = tuple([xp.empty((l,) + unit, dtype=dtype) for l in lengths])`?